### PR TITLE
fix(HTML Node): Prevent XSS in execution-data preview

### PR DESCRIPTION
**Pull request by netroy**
N8N-6496

### CODE REVIEW

File: packages/editor-ui/src/components/RunDataHtml.vue [hunk 1]
```diff
@@ -1,11 +1,25 @@
...
 import type { INodeExecutionData } from 'n8n-workflow';
 
+const sanitizeOptions: SanitizeOptions = {
+	allowVulnerableTags: false,
+	enforceHtmlBoundary: true,
+	disallowedTagsMode: 'discard',
+	allowedTags: [...defaults.allowedTags, 'style', 'img'],
+	allowedAttributes: {
+		'*': ['class'],
+	},
```
Comment (netroy) [inline packages/editor-ui/src/components/RunDataHtml.vue:17]:
```suggestion
	allowedAttributes: {
	    ...default.allowedAttributes,
		'*': ['class'],
	},
```

**Review by ivov — CHANGES_REQUESTED** [review]
CSS styling is lost!

Before:

image

After:

image

**Comment by netroy** [comment]
> CSS styling is lost!

That's because `enforceHtmlBoundary` is set to `true`, and the style tag in your example is in the wrong place. I explicitly made this decision to make the preview is consistent with what a user might see in a webview of apps like Outlook, etc.

Perhaps we should also update the placeholder to use semantically correct markup, by moving `style` and `script` tags to be inside the document.

**Comment by ivov** [comment]
> Perhaps we should also update the placeholder

Agree! We should either fix the placeholder to have effect or remove the CSS, to avoid confusing users. If removing it, we should run this by Giulio who requested the styling back when this was reviewed.

**Review by ivov — DISMISSED** [review]
1. Comparing side by side, `line-height`, `font-family` and `padding` are very different from the original. Also the HTML doc title was not rendered before, but now it is. 
2. The `Format HTML` option restores the old way of `script` and `style` at root level. We should update this or perhaps disable the option until we address it.

I do not mean to block a security fix on styling - up to you if this can be fixed in a follow-up.

**Comment by netroy** [comment]
I've reverted the `enforceHtmlBoundary` flag, and everything should now work.

The styles are still going to be slightly different, as the styles were not completely sanboxed earlier, and now they are. Which means the preview looks a lot more like what most users would see outside n8n.

**Review by ivov — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@0.234.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.234.0)

**Pull request merged**